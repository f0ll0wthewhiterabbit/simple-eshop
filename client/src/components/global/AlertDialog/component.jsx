import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'

const AlertDialog = ({
  isModalOpened,
  title,
  children,
  closeModal,
  storeFieldName,
  deleteItems,
}) => {
  const handleConfirmButton = () => {
    deleteItems(storeFieldName)
  }

  return (
    <Dialog
      open={isModalOpened}
      onClose={closeModal}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
      {children && (
        <DialogContent>
          <DialogContentText id="alert-dialog-description">{children}</DialogContentText>
        </DialogContent>
      )}
      <DialogActions>
        <Button onClick={closeModal} color="primary">
          Cancel
        </Button>
        <Button onClick={handleConfirmButton} color="primary" autoFocus>
          Confirm
        </Button>
      </DialogActions>
    </Dialog>
  )
}

AlertDialog.propTypes = {
  isModalOpened: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  closeModal: PropTypes.func.isRequired,
  storeFieldName: PropTypes.string,
  deleteItems: PropTypes.func.isRequired,
}

export default AlertDialog
