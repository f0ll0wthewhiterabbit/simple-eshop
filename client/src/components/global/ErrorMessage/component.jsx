import React from 'react'
import PropTypes from 'prop-types'
import PriorityHighOutlinedIcon from '@material-ui/icons/PriorityHighOutlined'

import { Wrapper, IconWrapper, Heading, Message } from './styles'

const ErrorMessage = ({ children, title }) => {
  return (
    <Wrapper>
      <IconWrapper>
        <PriorityHighOutlinedIcon />
      </IconWrapper>
      {title && (
        <Heading variant="h5" component="h1">
          {title}
        </Heading>
      )}
      {children && <Message variant="h6">{children}</Message>}
    </Wrapper>
  )
}

ErrorMessage.defaultProps = {
  children: '',
  title: '',
}

ErrorMessage.propTypes = {
  children: PropTypes.string,
  title: PropTypes.string,
}

export default ErrorMessage
