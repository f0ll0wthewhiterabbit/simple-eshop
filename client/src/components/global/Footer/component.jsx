import React from 'react'
import { Grid } from '@material-ui/core'
import PropTypes from 'prop-types'

import { MAIN_PAGE_PATH } from '../../../constants'
import {
  Wrapper,
  Info,
  SiteLink,
  SocialLinksWrapper,
  SocialLinksContainer,
  InfoWrapper,
  InfoHeading,
  InfoList,
  InfoItem,
  SocialLinksList,
  SocialLinksItem,
  SocialLink,
  SocialIconInstagram,
  SocialIconFacebook,
  SocialIconTwitter,
  SocialIconYoutube,
  FooterLink,
  Copyright,
} from './styles'

const Footer = ({ isAdmin, isAuthenticated }) => {
  return (
    <Wrapper>
      {!isAdmin && isAuthenticated && (
        <InfoWrapper maxWidth="lg">
          <Grid container spacing={3}>
            <Grid item xs={12} sm={6} md={3}>
              <InfoHeading variant="h4">About Us</InfoHeading>
              <InfoList>
                <InfoItem>About Us</InfoItem>
                <InfoItem>Community</InfoItem>
                <InfoItem>Jobs</InfoItem>
                <InfoItem>Shipping</InfoItem>
                <InfoItem>Contact Us</InfoItem>
              </InfoList>
            </Grid>
            <Grid item xs={12} sm={6} md={3}>
              <InfoHeading variant="h4">Customer Care</InfoHeading>
              <InfoList>
                <InfoItem>Search</InfoItem>
                <InfoItem>Privacy Policy</InfoItem>
                <InfoItem>2019 Lookbook</InfoItem>
                <InfoItem>Shipping &amp; Delivery</InfoItem>
                <InfoItem>Gallery</InfoItem>
              </InfoList>
            </Grid>
            <Grid item xs={12} sm={6} md={3}>
              <InfoHeading variant="h4">Our Services</InfoHeading>
              <InfoList>
                <InfoItem>Free Shipping</InfoItem>
                <InfoItem>Free Returnes</InfoItem>
                <InfoItem>Our Franchising</InfoItem>
                <InfoItem>Terms and conditions</InfoItem>
                <InfoItem>Privacy Policy</InfoItem>
              </InfoList>
            </Grid>
            <Grid item xs={12} sm={6} md={3}>
              <InfoHeading variant="h4">Information</InfoHeading>
              <InfoList>
                <InfoItem>Payment methods</InfoItem>
                <InfoItem>Times and shipping costs</InfoItem>
                <InfoItem>Product Returns</InfoItem>
                <InfoItem>Shipping methods</InfoItem>
                <InfoItem>Conformity of the products</InfoItem>
              </InfoList>
            </Grid>
          </Grid>
        </InfoWrapper>
      )}

      <SocialLinksWrapper>
        <SocialLinksContainer maxWidth="lg">
          <SocialLinksList>
            <SocialLinksItem>
              <SocialLink href="https://www.instagram.com/itechart_official">
                <SocialIconInstagram className="instagram" />
                <span className="social-links__title">Instagram</span>
              </SocialLink>
            </SocialLinksItem>
            <SocialLinksItem>
              <SocialLink href="https://twitter.com/itechart">
                <SocialIconTwitter className="twitter" />
                <span className="social-links__title">Twitter</span>
              </SocialLink>
            </SocialLinksItem>
            <SocialLinksItem>
              <SocialLink href="https://www.youtube.com/user/iTechArt">
                <SocialIconYoutube className="youtube" />
                <span className="social-links__title">Youtube</span>
              </SocialLink>
            </SocialLinksItem>
            <SocialLinksItem>
              <SocialLink href="https://www.facebook.com/iTechArt.Group">
                <SocialIconFacebook className="facebook" />
                <span className="social-links__title">Facebook</span>
              </SocialLink>
            </SocialLinksItem>
          </SocialLinksList>

          <Info variant="subtitle1" color="textSecondary" align="center" component="p">
            <FooterLink color="inherit" href="https://www.itechart.by/">
              iTechArt
            </FooterLink>
            &nbsp;Students Lab Vitebsk
          </Info>

          <Copyright>
            {'Â© '}
            <SiteLink to={MAIN_PAGE_PATH}>simple eShop</SiteLink> {new Date().getFullYear()}
          </Copyright>
        </SocialLinksContainer>
      </SocialLinksWrapper>
    </Wrapper>
  )
}

Footer.propTypes = {
  isAdmin: PropTypes.bool.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
}

export default Footer
