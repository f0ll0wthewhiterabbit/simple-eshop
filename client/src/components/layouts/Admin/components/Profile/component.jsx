import React from 'react'
import PropTypes from 'prop-types'
import Typography from '@material-ui/core/Typography'

import { Wrapper, ProfileAvatar } from './styles'

const Profile = ({ firstName, lastName }) => {
  const firstLetter = firstName.charAt(0).toUpperCase()
  const secondLetter = lastName.charAt(0).toUpperCase()

  return (
    <Wrapper>
      <ProfileAvatar>{`${firstLetter} ${secondLetter}`}</ProfileAvatar>
      <Typography variant="h6">{`${firstName} ${lastName}`}</Typography>
      <Typography variant="caption">Administrator</Typography>
    </Wrapper>
  )
}

Profile.propTypes = {
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
}

export default Profile
