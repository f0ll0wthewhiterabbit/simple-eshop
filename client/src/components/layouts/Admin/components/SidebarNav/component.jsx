import React from 'react'
import PropTypes from 'prop-types'
import PeopleIcon from '@material-ui/icons/People'
import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket'

import { ADMIN_USERS_PAGE_PATH, ADMIN_PRODUCTS_PAGE_PATH } from '../../../../../constants'
import { PagesList, NavigationLink, NavigationItem, PageIcon, PageText } from './styles'

const SidebarNav = ({ closeSidebar }) => {
  const pages = [
    {
      title: 'Users',
      href: ADMIN_USERS_PAGE_PATH,
      icon: <PeopleIcon />,
    },
    {
      title: 'Products',
      href: ADMIN_PRODUCTS_PAGE_PATH,
      icon: <ShoppingBasketIcon />,
    },
  ]

  const handleNavLinkClick = () => {
    closeSidebar()
  }

  return (
    <nav aria-label="navigation">
      <PagesList>
        {pages.map(page => (
          <NavigationItem key={page.title} button>
            <NavigationLink to={page.href} onClick={handleNavLinkClick} data-test="navigationLink">
              <PageIcon>{page.icon}</PageIcon>
              <PageText primary={page.title} />
            </NavigationLink>
          </NavigationItem>
        ))}
      </PagesList>
    </nav>
  )
}

SidebarNav.propTypes = {
  closeSidebar: PropTypes.func.isRequired,
}

export default SidebarNav
