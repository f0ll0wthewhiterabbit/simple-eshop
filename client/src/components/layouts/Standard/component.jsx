import React from 'react'
import PropTypes from 'prop-types'

import Header from '../../global/Header'
import Footer from '../../global/Footer'
import AlertDialog from '../../global/AlertDialog'
import { Root, Main } from './styles'

const StandardLayout = ({ children }) => {
  return (
    <Root>
      <AlertDialog title="Delete your account?">
        The remove request will be sent to the administrator. Unfortunately, your account will be
        deleted soon.
      </AlertDialog>
      <Header />
      <Main data-test="main">{children}</Main>
      <Footer />
    </Root>
  )
}

StandardLayout.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
}

export default StandardLayout
