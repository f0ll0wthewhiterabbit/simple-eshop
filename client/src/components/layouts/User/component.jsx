import React from 'react'
import PropTypes from 'prop-types'

import Header from '../../global/Header'
import Footer from '../../global/Footer'
import AlertDialog from '../../global/AlertDialog'
import { Root, Main } from './styles'

const UserLayout = ({ children }) => {
  return (
    <Root>
      <AlertDialog title="Delete your account?">
        The remove request will be sent to the administrator. Unfortunately, your account will be
        deleted soon.
      </AlertDialog>
      <Header />
      <Main>{children}</Main>
      <Footer />
    </Root>
  )
}

UserLayout.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
}

export default UserLayout
