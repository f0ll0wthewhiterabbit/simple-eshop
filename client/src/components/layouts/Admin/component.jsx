import React from 'react'
import PropTypes from 'prop-types'
import { withTheme } from 'styled-components'
import { useMediaQuery, Divider } from '@material-ui/core'

import Header from '../../global/Header'
import Footer from '../../global/Footer'
import Profile from './components/Profile'
import SidebarNav from './components/SidebarNav'
import AlertDialog from '../../global/AlertDialog'
import { Root, Wrapper, Sidebar, SidebarRoot, Main } from './styles'

const AdminLayout = ({ isSidebarOpened, theme, closeSidebar, children }) => {
  const isDesktop = useMediaQuery(theme.breakpoints.up('lg'), { defaultMatches: true })
  const shouldOpenSidebar = isDesktop ? true : isSidebarOpened

  const handleSidebarClose = () => {
    closeSidebar()
  }

  return (
    <Root>
      <AlertDialog title="Do you really want to delete selected items from database?" />
      <Header />
      <Wrapper>
        <Sidebar
          variant={isDesktop ? 'persistent' : 'temporary'}
          onClose={handleSidebarClose}
          open={shouldOpenSidebar}
          ModalProps={isDesktop ? {} : { keepMounted: true }} // Better open performance on mobile.
        >
          <SidebarRoot>
            <Profile />
            <Divider />
            <SidebarNav />
          </SidebarRoot>
        </Sidebar>
        <Main>{children}</Main>
      </Wrapper>
      <Footer />
    </Root>
  )
}

AdminLayout.propTypes = {
  isSidebarOpened: PropTypes.bool.isRequired,
  theme: PropTypes.shape({ breakpoints: PropTypes.object.isRequired }).isRequired,
  closeSidebar: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
}

export default withTheme(AdminLayout)
