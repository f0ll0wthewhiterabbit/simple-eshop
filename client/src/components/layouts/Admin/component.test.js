import React from 'react'
import { shallow } from 'enzyme'
import useMediaQuery from '@material-ui/core/useMediaQuery'

import { AdminLayout } from './component'
import findByTestAttr from '../../../utils/findByTestAttr'

jest.mock('@material-ui/core/useMediaQuery')

describe('AdminLayout component', () => {
  let props
  let wrapper
  const testContent = 'test content'
  const testChildren = <p>{testContent}</p>

  const generateWrapper = passedProps => {
    const initialProps = {
      children: testChildren,
      isSidebarOpened: true,
      closeSidebar: jest.fn(),
      theme: { breakpoints: { up: jest.fn() } },
    }
    props = { ...initialProps, ...passedProps }

    return shallow(<AdminLayout {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain correct children in main', () => {
    const main = findByTestAttr(wrapper, 'main')

    expect(main.find('p').text()).toEqual(testContent)
  })

  it('should contain opened sidebar', () => {
    const sidebar = findByTestAttr(wrapper, 'sidebar')

    expect(sidebar.props().open).toEqual(true)
  })

  it(`should contain opened sidebar if'isSidebarOpened=true' and width not desktop`, () => {
    useMediaQuery.mockImplementation(() => false)
    wrapper = generateWrapper()
    const sidebar = findByTestAttr(wrapper, 'sidebar')

    expect(sidebar.props().open).toEqual(true)
  })

  it(`shouldn't contain opened sidebar if prop 'isSidebarOpened=false' and width not desktop`, () => {
    useMediaQuery.mockImplementation(() => false)
    wrapper = generateWrapper({ isSidebarOpened: false })
    const sidebar = findByTestAttr(wrapper, 'sidebar')

    expect(sidebar.props().open).toEqual(false)
  })

  it(`should contain opened sidebar if prop 'isSidebarOpened=false' and width is desktop`, () => {
    useMediaQuery.mockImplementation(() => true)
    wrapper = generateWrapper({ isSidebarOpened: false })
    const sidebar = findByTestAttr(wrapper, 'sidebar')

    expect(sidebar.props().open).toEqual(true)
  })
})
// useMediaQuery.mockImplementation(() => false)
