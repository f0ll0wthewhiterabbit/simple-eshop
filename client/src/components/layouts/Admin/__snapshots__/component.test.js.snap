// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLayout component should render correctly 1`] = `
<styled.div>
  <Connect(AlertDialog)
    title="Do you really want to delete selected items from database?"
  />
  <Connect(Header) />
  <styled.div>
    <Styled(WithStyles(ForwardRef(Drawer)))
      ModalProps={
        Object {
          "keepMounted": true,
        }
      }
      data-test="sidebar"
      onClose={[Function]}
      open={true}
      variant="temporary"
    >
      <styled.div>
        <Connect(Profile) />
        <WithStyles(ForwardRef(Divider)) />
        <SidebarNav />
      </styled.div>
    </Styled(WithStyles(ForwardRef(Drawer)))>
    <styled.main
      data-test="main"
    >
      <p>
        test content
      </p>
    </styled.main>
  </styled.div>
  <Footer />
</styled.div>
`;
